contrib/endpoints/src/api_manager/context/request_context.cc

```diff
@@ -257,8 +259,12 @@ void RequestContext::FillReportRequestInfo(
   info->auth_audience = auth_audience_;
 
   if (!info->is_final_report) {
-    info->request_bytes = request_->GetGrpcRequestBytes();
-    info->response_bytes = request_->GetGrpcResponseBytes();
+    // Make sure we send delta metrics for intermediate reports.
+    info->request_bytes = request_->GetGrpcRequestBytes() - last_request_bytes;
+    info->response_bytes =
+        request_->GetGrpcResponseBytes() - last_response_bytes;
+    last_request_bytes += info->request_bytes;
+    last_response_bytes += info->response_bytes;
   } else {
     info->request_size = response->GetRequestSize();
     info->response_size = response->GetResponseSize();
```

Comment:
Have to handle final one too.  delta could be negative for the final one since it is getting from a different call. If negative, make it 0.